### ISSUE TITLE
When freezing TF-slim models: "Assign requires shapes of both tensors to match. lhs shape= [1,1,4096,1001] rhs shape= [1,1,4096,1000]" 

### ISSUE THREAD

**Issue by ShixuanZheng** [tensorflow/models#6933]
When using TF-slim, I download vgg_16/vgg_19/resnet_v1_50 from the site "https://github.com/tensorflow/models/tree/master/research/slim", and have already exported the inference graph. However, error was reported when running the command below:

python /home/zsx/.local/lib/python3.5/site-packages/tensorflow/python/tools/freeze_graph.py --input_graph=vgg_16_inf_graph.pb --input_checkpoint=vgg_16.ckpt --input_binary=true --output_graph=frozen_vgg_16.pb --output_node_names=softmax

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [1,1,4096,1001] rhs shape= [1,1,4096,1000]
	 [[Node: save/Assign_31 = Assign[T=DT_FLOAT, _class=["loc:@vgg_16/fc8/weights"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:GPU:0"](vgg_16/fc8/weights, save/RestoreV2/_3)]]

On vgg_19 and resnet_v1_50, the errors are the same: the shape's 4th dim of lhs is always 1 bigger than that of rhs. My tf version is 1.7.0.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Raphael-Hao** [comment]
same issue here

**Comment by ShixuanZheng** [comment]
Sorry for missing some field templates.
What is the top-level directory of the model you are using: N/A
Have I written custom code: No
 OS Platform and Distribution: Ubuntu 16.04.4 LTS
 TensorFlow installed from: pip
 TensorFlow version: 1.7.0
 Bazel version: N/A
 CUDA/cuDNN version: CUDA Version 9.0.176; cuDNN Main 7
 GPU model and memory: Tesla K80 x2, about 24G memory
 Exact command to reproduce: Cited above
The "vgg_16_inf_graph.pb" is generated using "export_inference_graph.py" provided by tf-slim.

**Comment by DreamerDoReMi** [comment]
Have you figured out this problem?

**Comment by Meresmata** [comment]
I also have the same problem. Tried it with vgg_16 

What is the top-level directory of the model you are using:N/A
Have I written custom code:No
OS Platform and Distribution:Windows10
TensorFlow installed from:anaconda
TensorFlow version:1.12
Bazel version:N/A
CUDA/cuDNN version:N/A
GPU model and memory:N/A
Exact command to reproduce:

freeze_graph.py --input_graph PATH\VGG_reverenz\vgg16_inf_graph.pb --input_checkpoint PATH\VGG_reverenz\vgg_16.ckpt --output_graph PATHVGG_reverenz\vgg_16_frozen_graph.pb --output_node_names vgg_16/fc8/squeezed:0 --input_binary True

The inputgraph was build with export_inference_graph.

in vgg.py the parameter num_classes is=1000 but in the pb files it's 10001. 

Crash happens at the line 128: 
```

      saver = saver_lib.Saver(
          var_list=var_list, write_version=checkpoint_version)
```

> Error:
> Assign requires shapes of both tensors to match. lhs shape= [1,1,4096,1001] rhs shape= [1,1,4096,1000]
> 	 [[node save/Assign_31 (defined at D:/USERNAME/ai2tfmodels/research/syntaxnet/tensorflow/tensorflow/python/tools/freeze_graph.py:128)  = Assign[T=DT_FLOAT, _class=["loc:@vgg_16/fc8/weights"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](vgg_16/fc8/weights, save/RestoreV2:31)]]

**Comment by Raphael-Hao** [comment]
@Meresmata Hi, have u figured out the problem?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
